wandb/sdk/wandb_run.py

```diff
@@ -3724,6 +3725,28 @@ def pb_to_dict(
                 logger.error("Error getting system metrics: %s", e)
         return {}
 
+    @property
+    def metadata(self) -> Metadata | None:
+        """Returns the metadata associated with this run.
+
+        Returns:
+            A dictionary of metadata.
+        """
+        if not self._backend or not self._backend.interface:
+            return None
+
+        handle = self._backend.interface.deliver_get_system_metadata()
```

Comment:
why do you want to have this, i think you want publish that allows users to modify the metadata 
isn't the idea to allow users to modify the default metadata?

Reply:
it's a draft, I have not finished yet. yes, I will have deliver, but in addition to this. i've thought of a "trick" to make it work nicely, just haven't yet coded it up.